core/keyboard_nav/ast_node.js

```diff
@@ -288,7 +288,7 @@ Blockly.ASTNode.prototype.findPreviousEditableField_ = function(location,
   var previousField = null;
   var startIdx = opt_last ? fieldRow.length - 1 : fieldIdx - 1;
   for (var i = startIdx, field; field = fieldRow[i]; i--) {
-    if (field.isCurrentlyEditable()) {
```

Comment:
Why this change?

Reply:
Because isCurrentlyEditable will always be false for fields when in readOnly mode. So the user wouldn't be able to navigate to the fields.